### PR TITLE
[flutter_markdown] Fix shrinkWrap not taken into account with single child

### LINKED ISSUE #105299: [flutter_markdown] MarkdownBody.shrinkWrap not taken into account with single child
`MarkdownBody`'s `shrinkWrap` property is not taken into account when the list of children provided by `MarkdownWidget` has only one child.

The reason seems to be that:
1. When there's only one child `MarkdownBody` doesn't wrap it into a `Column`:

https://github.com/flutter/packages/blob/4b2114beb992da0e031c4e9f94e04f6e659e704b/packages/flutter_markdown/lib/src/widget.dart#L443-L453

2. And the inner column that is created by `MarkdownBuilder ` doesn't specify any `mainAxisSize`. 

https://github.com/flutter/packages/blob/4b2114beb992da0e031c4e9f94e04f6e659e704b/packages/flutter_markdown/lib/src/builder.dart#L366-L371

## Steps to Reproduce

```dart
@override
Widget build(BuildContext context) {
  return Stack(
    children: const <Widget>[
      Align(
        alignment: Alignment.bottomCenter,
        child: MarkdownBody(
          data: 'This is a [link](https://flutter.dev/)',
          shrinkWrap: true,
        ),
      ),
    ],
  );
}
```

**Actual results:** The widget takes all the available height.
image

**Expected results:** The widget should wrap the markdown content.
image

</details>

<details>
  <summary>Logs</summary>

```
❯ flutter doctor -v
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.4 21F79 darwin-x64, locale en-NL)
    • Flutter version 3.0.1 at /Users/davidmigloz/repos/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (2 weeks ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/davidmigloz/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • ANDROID_HOME = /Users/davidmigloz/Library/Android/sdk
    • Java binary at: /Users/davidmigloz/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8512546/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/davidmigloz/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8512546/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.1)
    • Android Studio at /Users/davidmigloz/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/211.7628.21.2111.8309675/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.1)
    • IntelliJ at /Users/davidmigloz/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.1)
    • IntelliJ at /Users/davidmigloz/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5591.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at /Users/davidmigloz/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.67.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (3 available)
    • Pixel 3 (mobile) • 8ANX0VZ7Q • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos     • darwin-x64     • macOS 12.4 21F79 darwin-x64
    • Chrome (web)     • chrome    • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

### PR DESCRIPTION
**Pull request by davidmigloz**
Fixes https://github.com/flutter/flutter/issues/105299

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@domesticmouse Can you advise on tests here?

@davidmigloz Please merge in the last master so CI re-runs; the failures are due to a repo-wide failure that has been resolved.

**Review by domesticmouse — CHANGES_REQUESTED** [review]

**Comment by davidmigloz** [comment]
I've added some tests and a demo page in the example app.
Let me know if they are good enough.

shrink_wrap

File: packages/flutter_markdown/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.6.11
+
+* Fix shrinkWrap not taken into account with single child #105299
```
Comment (domesticmouse) [inline packages/flutter_markdown/CHANGELOG.md:3]: I'm guessing the `#105299` at end of line is referring to the bug fixed by this change. I'd personally prefer this to be `flutter/flutter#105299` or even `flutter/flutter Issue #105299` as that is not ambiguous.

**Review by domesticmouse — APPROVED** [review]

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (davidmigloz) [reply packages/flutter_markdown/CHANGELOG.md]: I've added a link and prefixed it with: [https://github.com/flutter/flutter/issues/105299](https://github.com/flutter/flutter/issues/105299) (same syntax GitHub uses).

I've also rewritten the message to use a "present tense indicative for verbs" as indicated in the [changelog style guidelines](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style).

**Review by domesticmouse — APPROVED** [review]

File: packages/flutter_markdown/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.6.11
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:1]: Since this is a bugfix, it should be 0.6.11+3.

**Review by stuartmorgan-g — CHANGES_REQUESTED** [review]
One note on the version, but otherwise this looks good to land.

File: packages/flutter_markdown/CHANGELOG.md (hunk 2 above)
Comment (davidmigloz) [reply packages/flutter_markdown/CHANGELOG.md]: I guess you meant `0.6.10+3`. Fixed!

**Review by stuartmorgan-g — APPROVED** [review]

**Pull request merged**